typescript-practice/src/ts/views/product-view.ts

```diff
@@ -0,0 +1,320 @@
+  addBtn: HTMLButtonElement;
+  closeBtn: HTMLButtonElement;
+  removeSelected: HTMLButtonElement;
+
+  formInput: NodeListOf<HTMLInputElement>;
+  nameInput: HTMLInputElement;
+  priceInput: HTMLInputElement;
+  imageInput: HTMLInputElement;
+  imageProduct: HTMLInputElement;
+  desInput: HTMLInputElement;
+
+  constructor() {
+    this.formModal = this.getElement('#formModal');
+    this.modal = this.getElement('#modal');
+    this.heading = this.getElement('#form-heading');
+    this.productList = this.getElement('.product-list');
+
+    this.formInput = this.formModal.querySelectorAll('.form-input');
+
+    this.addBtn = this.getButtonElement('#open-form');
+    this.closeBtn = this.getButtonElement('#close-form');
+    this.saveProduct = this.getButtonElement('#save-product');
+    this.removeSelected = this.getButtonElement('#remove-selected');
+
+    this.nameInput = this.getInputElement('#product-name');
+    this.priceInput = this.getInputElement('#product-price');
+    this.imageInput = this.getInputElement('#product-image');
+    this.imageProduct = this.getInputElement('#show-image');
+    this.desInput = this.getInputElement('#product-des');
+  }
+
+  // Retrieve an element from the DOM
+  getElement(selector: string): HTMLElement {
+    return document.querySelector(selector) as HTMLElement;
+  }
+
+  // Get Input element
+  getInputElement(selector: string): HTMLInputElement {
+    return document.querySelector(selector) as HTMLInputElement;
+  }
+
+  // Get Button element
+  getButtonElement(selector: string): HTMLButtonElement {
+    return document.querySelector(selector) as HTMLButtonElement;
+  }
+
+  // Reset input to ''
+  _resetInput() {
+    this.nameInput.value = '';
+    this.priceInput.value = '';
+    this.desInput.value = '';
+    this.imageInput.value = '';
+    this.imageProduct.style.display = 'none';
+    this.imageProduct.src = '';
+    this.saveProduct.value = '';
+  }
+
+  // Display form modal
+  openForm() {
+    this.modal.style.display = 'block';
+  }
+
+  // Bind open form modal
+  bindOpenform() {
+    this.addBtn.addEventListener('click', (e) => {
+      this.heading.innerText = 'Add New Product';
+      this.openForm();
+      this._resetInput();
+    });
+  }
+
+  // Hide form modal
+  closeForm() {
+    this.modal.style.display = 'none';
+  }
+
+  // Bind close form modal
+  bindCloseform() {
+    this.closeBtn.addEventListener('click', (e) => {
+      e.preventDefault();
+      this.saveProduct.value = '';
+      this.closeForm();
+    });
+    this.modal.addEventListener('click', (e) => {
+      if (e.target == e.currentTarget) this.closeForm();
+    });
+  }
+
+  // Validate input in form modal
+  validateInput() {
+    if (this.formInput) {
+      this.formInput.forEach(function (inputElement) {
+        if (inputElement) {
+          const formGroup = inputElement.parentElement as HTMLElement;
+          const errorElement = formGroup.querySelector('.form-message') as HTMLElement;
```

Comment:
Why don't you get error message element by ID?

Reply:
I want to get error message element under each input in form-group so I use class

Reply:
Please assign an meaningful ID for each error message element.